pkg/apiserver/apiserver.go

```diff
@@ -311,24 +311,32 @@ func isPrettyPrint(req *http.Request) bool {
 
 // writeJSON renders an object as JSON to the response.
 func writeJSON(statusCode int, codec runtime.Codec, object runtime.Object, w http.ResponseWriter, pretty bool) {
+	w.Header().Set("Content-Type", "application/json")
+	// We send the status code before we encode the object, so if we error, the status code stays but there will
+	// still be an error object.  This seems ok, the alternative is to validate the object before
+	// encoding, but this really should never happen, so it's wasted compute for every API request.
+	w.WriteHeader(statusCode)
+	if pretty {
+		prettyJSON(codec, object, w)
```

Comment:
Oh, you're right, it doesn't. I forgot, I've wanted such a thing before,
too.

On Fri, Sep 18, 2015 at 12:29 PM, Brendan Burns <notifications@github.com>
wrote:

> In pkg/apiserver/apiserver.go
> <https://github.com/kubernetes/kubernetes/pull/14148#discussion_r39892213>
> :
>
> > @@ -311,24 +311,32 @@ func isPrettyPrint(req *http.Request) bool {
> >
> >  // writeJSON renders an object as JSON to the response.
> >  func writeJSON(statusCode int, codec runtime.Codec, object runtime.Object, w http.ResponseWriter, pretty bool) {
> > +	w.Header().Set("Content-Type", "application/json")
> > +	// We send the status code before we encode the object, so if we error, the status code stays but there will
> > +	// still be an error object.  This seems ok, the alternative is to validate the object before
> > +	// encoding, but this really should never happen, so it's wasted compute for every API request.
> > +	w.WriteHeader(statusCode)
> > +	if pretty {
> > +		prettyJSON(codec, object, w)
>
> interestingly, json.Encoder doesn't seem to have a pretty print option.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14148/files#r39892213>.
>